### PR TITLE
Firefly-1037, Firefly-1038: multiple issues

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1037](https://jira.ipac.caltech.edu/browse/FIREFLY-1037), [Firefly-1038](https://jira.ipac.caltech.edu/browse/FIREFLY-1038): Multiple Issues

 - Firefly-1037: added ipac and caltech logo to version dialog
 - Firefly-1038: fixed stretch so all planes of the cube are changed
 - Other: show checkbox on the stretch dropdown if that stretch is selected
 - Other: improved the heuristic to determine is images are noisy

#### Testing
   - https://fireflydev.ipac.caltech.edu/firefly-1037-1038/firefly/
   - Look at help dialog
   - Load a cube, change the stretch using the dropdown, the stretch should be the same on all planes
       - confirm by looking at the stretch on the dropdown, it should show a checkbox

### CODE REVIEW

**Comment by lrebull** [comment]
you have "look at help dialog" -- did you grab the most recent help that i checked into git earlier today?

**Comment by lrebull** [comment]
(I literally just checked in another version with Gregory's latest tweak, but I'll take anything you've done that grabs the help as of ~6pm yesterday.)

File: src/firefly/js/ui/VersionInfo.jsx [hunk 1]
```diff
@@ -49,28 +52,47 @@ function VersionInfoFull() {
...
+                <div className='Version-grid-warning'>
+                    {isVersionPreRelease() ?
+                        `Warning: Early preview of Firefly ${getDevCycle()}` :
+                        `Warning: Development build of Firefly on dev cycle ${getDevCycle()}`
+                    }
+                </div>
+            }
+            <div style={{padding:'10px 5px 3px 5px', width:600, fontSize:'smaller'}}>
+                <span>
+                    Firefly development at IPAC has been supported by NASA, principally through IRSA,
```
Comment (gpdf) [inline src/firefly/js/ui/VersionInfo.jsx:86]: Can we add the following hyperlinks, please:

- IPAC = https://ipac.caltech.edu/
- IRSA = https://irsa.ipac.caltech.edu/
...

File: src/firefly/js/ui/VersionInfo.jsx [hunk 2]
```diff
@@ -49,28 +52,47 @@ function VersionInfoFull() {
...
+                    and by the National Science Foundation, through the Vera C. Rubin Observatory.
```
Comment (gpdf) [inline src/firefly/js/ui/VersionInfo.jsx:87]: and

- Vera C. Rubin Observatory = https://www.lsst.org/

Thanks!

File: src/firefly/js/ui/VersionInfo.jsx [hunk 3]
```diff
@@ -49,28 +52,47 @@ function VersionInfoFull() {
...
+                    <a href='https://www.ipac.caltech.edu' target='ipac-window'><img alt='IPAC' src={IPAC_LOGO}/></a>
+                </div>
+                <div style={{display: 'flex', justifyContent: 'space-evenly', marginLeft:60, alignItems:'flex-start'}}>
+                    <div className='Version-grid'>
+                        <Entry desc={VER} value={version}/>
+                        <Entry desc={BUILT_ON} value={BuildTime}/>
+                        <Entry desc={COMMIT} value={BuildCommit}/>
+                        {major ? <Entry desc={FF_LIB} value={getFireflyLibraryVersionStr()}/> : ''}
+                        {ffCommit && <Entry desc={FF_COM} value={ffCommit}/>}
+                        {ffTag && <Entry desc={FF_TAG} value={ffTag}/>}
```
Comment (gpdf) [inline src/firefly/js/ui/VersionInfo.jsx:69]: I'm not quite sure how the copy-to-clipboard icon in this dialog works, but I just want to be sure that it only copies the version info and not the acknowledgement sentence.  (I don't have VPN access at the moment to test it live.)

**Review by gpdf — CHANGES_REQUESTED** [review]
Small suggestions for changes.

Perhaps add a note in the PR that it addresses FIREFLY-1029 (the "show checkmark on the stretch dropdown" item)?

**Comment by robyww** [comment]
Screen Shot 2022-07-27 at 3 40 59 PM

File: src/firefly/js/ui/VersionInfo.jsx (hunk 3 above)
Comment (robyww) [reply src/firefly/js/ui/VersionInfo.jsx:69]: It only copies the version. Nothing has changed.

Comment (ejoliet) [reply src/firefly/js/ui/VersionInfo.jsx:69]: yes it does only copy the version.

File: src/firefly/js/ui/VersionInfo.jsx [hunk 4]
```diff
@@ -49,28 +52,47 @@ function VersionInfoFull() {
...
+            }
+            <div style={{padding:'10px 5px 3px 5px', width:600, fontSize:'smaller'}}>
+                <span>
+                    Firefly development at IPAC has been supported by NASA, principally through IRSA,
+                    and by the National Science Foundation, through the Vera C. Rubin Observatory.
+                    Firefly is open-source software, available on&nbsp;
+                </span>
+                <a href="https://github.com/Caltech-IPAC/firefly" target='github-window'>GitHub</a>
+                <span> and </span>
+                <a href="https://hub.docker.com/repository/docker/ipac/firefly" target='dockerhub-window'>DockerHub</a>
```
Comment (gpdf) [inline src/firefly/js/ui/VersionInfo.jsx:92]: There's a missing period after the `DockerHub</a>`.

File: src/firefly/js/ui/VersionInfo.jsx (hunk 3 above)
Comment (gpdf) [reply src/firefly/js/ui/VersionInfo.jsx:69]: Thanks!

**Comment by gpdf** [comment]
The screenshot looks good, if the additional hyperlinks and '.' are added.  Thank you!

**Review by ejoliet — APPROVED** [review]
Stretch tested. All good! 
 thanks!

**Comment by ejoliet** [comment]
> you have "look at help dialog" -- did you grab the most recent help that i checked into git earlier today?
@lrebull it's the version help dialog at the lower right-hand corner of the app.

**Comment by gpdf** [comment]
In many applications the version dialog is also available by clicking on the application logo in the upper left corner.

**Comment by lrebull** [comment]
everything I checked seemed to be ok. (even including the online help! :) )

**Comment by robyww** [comment]
@gpdf I did not know there was a ticket for check boxes. I will add it

File: src/firefly/js/ui/VersionInfo.jsx (hunk 1 above)
Comment (robyww) [reply src/firefly/js/ui/VersionInfo.jsx:86]: ok

File: src/firefly/js/ui/VersionInfo.jsx (hunk 4 above)
Comment (robyww) [reply src/firefly/js/ui/VersionInfo.jsx:92]: fixed

**Pull request merged**